contracts/cosmwasm-vm/cw-xcall/src/execute_call.rs

```diff
@@ -81,18 +81,23 @@ impl<'a> CwCallService<'a> {
         let (response, event) = match msg.result {
             cosmwasm_std::SubMsgResult::Ok(_res) => {
```

Comment:
Cant comment so high but when you reuse the request id db and remove it on reply does this not break when the call then returns?

Reply:
ya .. it is reused in reply so we shouldnt be removing it on reply . We can prevent reply spamming by checking if callReply state is not null. Let me fix this.